### ISSUE TITLE
Possible drop in? For other addons?

### ISSUE THREAD

**Issue by Noitidart** [carlin-q-scott/browser-media-keys#29]
I wanted to create a hotkey global hotkey for my addon. Would it be easy for you to make a distributable of the workers so other addon devs could easily setup global hotkeys in their addons? :)

This would be a huge one up on firefox addons versus chrome addons, as google chrome addons can only do numpad for global hotkeys. :d

**Comment by carlin-q-scott** [comment]
I guess it depends on what you want to do with it. As is, you could just copy the windowsHotkeys.js file in /data and modify it to look for different keys. I do like the idea of sharing this functionality in a way that could let it grow on its own but I'm not sure how it would be packaged.

**Comment by Noitidart** [comment]
I would think put it all into a worker, then we would just offer a worker people can import. :)

I would think something that works with PromiseWorker, that has real easy communication: https://developer.mozilla.org/en-US/docs/Web/API/PromiseWorker

**Comment by carlin-q-scott** [comment]
It took me a while to figure out the benefit of returning a promise in this situation. Providing a promise would be helpful for the registration of the key combos but the regular message passing interface seems better suited for key input events.

My main concern was how would someone actually import the code into their plugin. Do people use node packages for that?

**Comment by Noitidart** [comment]
Yep don't worry about the droppable. If you can knock out the OS X portion (Ill of course help out), Ill make this 1) an npm module for JPM/SDK addons and 2) a promise worker drop in for overlay/bootstrap addons. :)

**Comment by Noitidart** [comment]
Here's some documentation on it: https://developer.mozilla.org/en-US/Add-ons/SDK/Tutorials/Using_third-party_modules_%28jpm%29

And here is a droppable module a friend of mine made: https://github.com/freaktechnik/jetpack-panelview/blob/master/docs/panelview.md

Cool stuff :)

**Comment by Noitidart** [comment]
Hey @carlin-q-scott my screenshot addon got approved and I wanted to add in global hotkey support. For windows the method you used here isnt limited to media keys so Ill use that. For mac ill use RegisterHotKey.

But for Unix I had a question, it seems your method is specific to the media keys is this true?

**Comment by carlin-q-scott** [comment]
Yeah, it's my understanding that the Linux code is media key centric and not too universal unfortunately. Someone else wrote it but I reviewed it at least.
